data_prep/mp_prep_other_inputs_annual.py

```diff
@@ -38,140 +43,194 @@ def mp_prep_other_inputs(tile_id_list):
+
+        ### Drivers of tree cover loss processing
+        uu.print_log("STEP 1: Preprocess drivers of tree cover loss")
+
+        uu.s3_file_download(os.path.join(cn.drivers_raw_dir, cn.pattern_drivers_raw), cn.docker_tile_dir, sensit_type)
+
+        # Creates tree cover loss driver tiles.
+        # The raw driver tile should have NoData for unassigned drivers as opposed to 0 for unassigned drivers.
+        # For the 2020 driver update, I reclassified the 0 values as NoData in ArcMap. I also unprojected the global drivers
+        # map to WGS84 because running the homolosine projection that Jimmy provided was giving incorrect processed results.
+        source_raster = cn.pattern_drivers_raw
+        out_pattern = cn.pattern_drivers
+        dt = 'Byte'
         if cn.count == 96:
-            processes = 50  # 60 processors = >730 GB peak (for European natural forest forest removal rates); 50 = XXX GB peak
-            uu.print_log(f'Checking for empty tiles of {output_pattern} pattern with {processes} processors...')
-            pool = multiprocessing.Pool(processes)
-            pool.map(partial(uu.check_and_delete_if_empty, output_pattern=output_pattern), tile_id_list)
-            pool.close()
-            pool.join()
-        elif cn.count <= 2: # For local tests
-            processes = 1
-            uu.print_log(f'Checking for empty tiles of {output_pattern} pattern with {processes} processors using light function...')
-            pool = multiprocessing.Pool(processes)
-            pool.map(partial(uu.check_and_delete_if_empty_light, output_pattern=output_pattern), tile_id_list)
-            pool.close()
-            pool.join()
+            processes = 87  # 45 processors = 70 GB peak; 70 = 90 GB peak; 80 = 100 GB peak; 87 = 125 GB peak
         else:
-            processes = int(cn.count / 2)
-            uu.print_log(f'Checking for empty tiles of {output_pattern} pattern with {processes} processors...')
-            pool = multiprocessing.Pool(processes)
-            pool.map(partial(uu.check_and_delete_if_empty, output_pattern=output_pattern), tile_id_list)
-            pool.close()
-            pool.join()
-        uu.print_log("\n")
-
-
-    # Uploads output tiles to s3
-    for i in range(0, len(output_dir_list)):
-        uu.upload_final_set(output_dir_list[i], output_pattern_list[i])
+            processes = int(cn.count/2)
+        uu.print_log("Creating tree cover loss driver tiles with {} processors...".format(processes))
+        pool = multiprocessing.Pool(processes)
+        pool.map(partial(uu.mp_warp_to_Hansen, source_raster=source_raster, out_pattern=out_pattern, dt=dt),
+                 tile_id_list)
+        pool.close()
+        pool.join()
+
+        for output_pattern in [
+            cn.pattern_drivers
+        ]:
+
+            if cn.count == 96:
+                processes = 50  # 60 processors = >730 GB peak (for European natural forest forest removal rates); 50 = XXX GB peak
```

Comment:
Delete the comment. I doubt that the note on 60 processors for Europe is correct, and it's not relevant for this driver processing call.

Reply:
Changes made in the following commit: b359551c7aa4405098e8f07ea7a582c6eddb963c